src/Annotations/Field.php

```diff
@@ -4,27 +4,55 @@
 
 namespace TheCodingMachine\GraphQLite\Annotations;
 
+use Doctrine\Common\Annotations\Annotation\Attribute;
+
 /**
  * @Annotation
- * @Target({"METHOD"})
+ * @Target({"PROPERTY", "METHOD"})
  * @Attributes({
  *   @Attribute("name", type = "string"),
  *   @Attribute("outputType", type = "string"),
  *   @Attribute("prefetchMethod", type = "string"),
+ *   @Attribute("for", type = "string[]"),
+ *   @Attribute("description", type = "string"),
+ *   @Attribute("inputType", type = "string"),
  * })
  */
 class Field extends AbstractRequest
 {
     /** @var string|null */
     private $prefetchMethod;
 
+    /**
+     * Input/Output type names for which this fields should be applied to.
+     *
+     * @var string[]|null
+     */
+    private $for = null;
+
+    /**
+     * @var string|null
+     */
+    private $description;
```

Comment:
Usually PHPDoc comments work out well for API documentation, but I believe it's not always like this. I think the "description" field can provide more flexibility. Here's some use cases:

- As you mentioned there might be a situation where you'd like to have different description per different input/type. You might have different validation rules or security restrictions and it would be a great idea mentioning such things in the API. Let's say I have "email" field in user class with 3 different types:
  - `CreateUserInput` - used on sign up. It's required for a user to specify their email
  - `UpdateUserInput` - only admin user can update "email" field
  - `User` - only admin can see emails or users can see only their emails (we don't want to expose emails to other users for example)

  I forgot to mention it above, but I've adjusted `@Fields` so it would be possible to add several of them for the same property. Like this:
```
/**
 * @GQL\Input(name="CreateUserInput")
 * @GQL\Input(name="UpdateUserInput", update=true)
 * @GQL\Type()
 */
class User {

    /**
     * @var string User email
     *
     * @GQL\Field(for="CreateUserInput", description="User email.")
     * @GQL\Field(for="UpdateUserInput", description="User email. Only admin can update it.")
     * @GQL\Field(for="User", description="User email. Can be accessed by an admin or currently logged in user to see only their emails.")
     */
    public string $email;
}
```
- If you apply `@Field` annotation on a method, the comment might not be appropriate. PHPDoc comments for methods usually start with a verb ( like "Return", "Get" ). I don't think field description should start with a verb, because they do not perform any actions. So instead of "Get user email", the correct description should be "User email". `description` attribute can help fixing that. 
- Sometimes you might want to leave some technical comment in PHPDoc, which should be visible only for internal developers and hidden from API documentation. In `description` attribute you can specify only documentation needed for the API.

Does this make sense to you?

P. S. I realized that I didn't adjust this `description` field to work with `@Type`, it works only with `@Input`. If it makes sense, I'll fix that.

Reply:
Ok, your explanation makes a lot of sense to me. We will just have to make sure we correctly document it :+1:

Reply:
Fixed Field description to work with `@Type`